pkg/table/index.go

```diff
@@ -32,46 +32,54 @@ type IndexIterator interface {
 // CreateIdxOpt contains the options will be used when creating an index.
 type CreateIdxOpt struct {
 	commonMutateOpt
-	IgnoreAssertion bool
-	FromBackFill    bool
+	ignoreAssertion bool
+	fromBackFill    bool
 }
 
 // NewCreateIdxOpt creates a new CreateIdxOpt.
 func NewCreateIdxOpt(opts ...CreateIdxOption) *CreateIdxOpt {
 	opt := &CreateIdxOpt{}
 	for _, o := range opts {
-		o.ApplyCreateIdxOpt(opt)
+		o.applyCreateIdxOpt(opt)
 	}
 	return opt
 }
 
+// IgnoreAssertion indicates whether to ignore assertion.
+func (opt *CreateIdxOpt) IgnoreAssertion() bool {
+	return opt.ignoreAssertion
+}
+
+// FromBackFill indicates whether the index is created by DDL backfill worker.
+func (opt *CreateIdxOpt) FromBackFill() bool {
+	return opt.fromBackFill
+}
+
 // CreateIdxOption is defined for the Create() method of the Index interface.
 type CreateIdxOption interface {
-	ApplyCreateIdxOpt(*CreateIdxOpt)
+	applyCreateIdxOpt(*CreateIdxOpt)
 }
 
-// CreateIdxOptFunc is defined for the Create() method of Index interface.
-// Here is a blog post about how to use this pattern:
-// https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
-type CreateIdxOptFunc func(*CreateIdxOpt)
```

Comment:
Also delete `CreateIdxOptFunc` to avoid some operations like `WithIgnoreAssertion(createIdxOpt)` to modify the internal state of `CreateIdxOpt`.